src/Medidata.ZipkinTracer.Core/Helpers/ParserHelper.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Globalization;
+
+namespace Medidata.ZipkinTracer.Core.Helpers
+{
+    internal static class ParserHelper
+    {
+        /// <summary>
+        /// Checks if string value can be converted to 128bit (Guid) or 64bit (long)
+        /// </summary>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        internal static bool IsParsableTo128Or64Bit(this string value)
+        {
+            if (value.IsParsableToGuid()) return true;
+            return value.IsParsableToLong();
+        }
+
+        /// <summary>
+        /// Checks if hex string value is parsable to Guid
+        /// </summary>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        internal static bool IsParsableToGuid(this string value)
+        {
+            Guid result;
+            return Guid.TryParse(value, out result);
```

Comment:
make sure this is normal hex format (16 or 32 char) as opposed to UUID format (with hyphens), In general, I'd avoid GUID as usually it implies the latter (which will break people)